### PR TITLE
Add test that VM is not stronger than regular conversion on empty sprop inductive

### PR DESCRIPTION
**Pull request by SkySkimmer**
There is a risk of forgetting this if we ever implement proof irrelevance in the VM so let's add a test.

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]

File: test-suite/bugs/bug_19041.v [hunk 1]
```diff
@@ -4,4 +4,6 @@ Goal forall (x:T true) (y:T false), match x return bool with end = match y retur
 Proof.
   intros.
   Fail reflexivity.
+  match goal with |- ?a = ?b => vm_cast_no_check (eq_refl a) end.
+  Fail Qed.
```
Comment (ppedrot) [inline test-suite/bugs/bug_19041.v:8]: This is a bit ugly, you could have created another proof, but this is not a strong RFC.

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@coqbot merge now
